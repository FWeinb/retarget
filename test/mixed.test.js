import retarget from '../src/index'

test('first use tagged template', () => {
  const selector = retarget`profile.name.${retarget.last}`

  const STATE = {
    profile: {
      name: {
        first: 'Waylon',
        last: 'Jennings'
      }
    }
  }

  expect(selector(STATE)).toMatchSnapshot()
})

test('first use proxy', () => {
  const selector = retarget.profile`name.${retarget.last}`

  const STATE = {
    profile: {
      name: {
        first: 'Waylon',
        last: 'Jennings'
      }
    }
  }

  expect(selector(STATE)).toMatchSnapshot()
})
