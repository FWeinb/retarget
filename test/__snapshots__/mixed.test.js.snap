// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first use proxy 1`] = `"Jennings"`;

exports[`first use tagged template 1`] = `"Jennings"`;
